### PR TITLE
Virtualmachines dev rosanto

### PR DESCRIPTION
**Pull request by rodrigosantosms**
# Overview/Summary

Adding recommendations and queries for Virtual Machines

## This PR adds

1. Recommendations
2. Queries

## As part of this Pull Request I have

- [ x] Read the [Contribution Guide](https://azure.github.io/Azure-Proactive-Resiliency-Library/contributing) and ensured this PR is compliant with the guide
- [x ] Checked for duplicate [Pull Requests](https://github.com/Azure/Azure-Proactive-Resiliency-Library/pulls)
- [ x] Associated it with relevant [GitHub Issues](https://github.com/Azure/Azure-Proactive-Resiliency-Library/issues) or ADO Work Items (Internal Only)
- [ x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Azure-Proactive-Resiliency-Library/tree/main)
- [x ] Ensured PR tests are passing
- [x] Updated relevant and associated documentation (e.g. Contribution Guide, Docs etc.)

### CODE REVIEW

File: docs/content/services/compute/virtual-machines/_index.md [hunk 1]
```diff
@@ -1,121 +1,645 @@
...
 ## Summary of Recommendations
 
-The below table shows the list of resiliency recommendations for Virtual Machines and associated resources.
-
-{{< alert style="info" >}}
-
-Use the table of contents (TOC) navigation menu on the right to jump to the recommendation you wish to review.
-
-{{< /alert >}}
-
```
Comment (jtracey93) [inline docs/content/services/compute/virtual-machines/_index.md:21]: Can we keep this. I think it's important to call out as it's useful.

File: docs/content/services/compute/virtual-machines/_index.md [hunk 2]
```diff
@@ -1,121 +1,645 @@
...
+| vm15 | VirtualMachines | Customer DNS Servers should be configured in the Virtual Network level | No |
+| vm16 | VirtualMachines | Private IP Address should be configured as Static | No |
+| vm17 | VirtualMachines | Shared disks should only be enabled in Clustered servers | No |
+| vm18 | VirtualMachines | The Network access to the VM disk is set to "Enable Public access from all networks" | No |
+| vm19 | VirtualMachines | Virtual Machine is not compliant with Azure Policies | No |
+| vm20 | VirtualMachines | Enable disk encryption, Enable data at rest encryption by default | No |
+| vm21 | VirtualMachines | Enable Insights to get more visibility into the health and performance of your virtual machine | No |
+| vm22 | VirtualMachines | Diagnostic Settings should be configured for all Azure Resources | No |
+| vm23 | VirtualMachines | Tags are inconsistent across Virtual Machines | No |
+| vm24 | VirtualMachines | Tag shows incorrect value for the Availability Zone number for a Virtual Machine | No |
```
Comment (jtracey93) [inline docs/content/services/compute/virtual-machines/_index.md:40]: Can we uppercase the IDs on the recommendations to be the same as the headings below. 

E.g. `vm01` should be `VM01`

Comment (jtracey93) [inline docs/content/services/compute/virtual-machines/_index.md:40]: Let's also make all of the IDs a link to the relevant place on the page via a tag link using `#<tag-heading>`

**Review by jtracey93 — CHANGES_REQUESTED** [review]
A few changes please mate. Also make these in the template as well so all going forward use it. 👍

File: docs/content/services/compute/virtual-machines/_index.md (hunk 2 above)
Comment (rodrigosantosms) [reply docs/content/services/compute/virtual-machines/_index.md:40]: good catch

**Pull request merged**